File: ScreenAdaptation/app/src/main/java/com/tangpj/screenadaptation/MainActivity.java
package com.tangpj.screenadaptation;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView tvSw;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        int sw = getResources().getConfiguration().smallestScreenWidthDp;
        tvSw = findViewById(R.id.tv_sw);
        tvSw.setText("sw = " + sw + "dp");
    }
}

File: SimpleCalces/app/src/main/java/com/tangpj/simplecalces/MainActivity.kt
package com.tangpj.simplecalces

import android.content.Intent
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_library1.setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW,Uri.parse("simple://library1"))
            startActivity(intent)
        }
        btn_library2.setOnClickListener{
            val intent = Intent(Intent.ACTION_VIEW,Uri.parse("simple://library2"))
            startActivity(intent)
        }

    }
}
